streams/src/main/java/org/apache/kafka/streams/processor/StateRestoreListener.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.processor;
+
+/**
+ * Class for listening to various states of the restoration process of a StateStore.
+ *
+ * When using a StateRestoreListener for reporting all state store recovery progress in an application,
```

Comment:
Better state "when calling `setState...` in \@code KafkaStreams, the passed instance is expected to be stateless since.."

Because not everyone understand what does "... for reporting all state store recovery.." means, stating from the API point of view would be easier to understand. Ditto elsewhere.